### PR TITLE
feat(curriculum): add business card lab

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

As this is the first CSS lab, the user stories are not open-ended in terms of design.
I have asked campers to add specific colours, margins, and borders.

To do:
- [ ] Add demo project link.

### CODE REVIEW

**Review by moT01 — COMMENTED** [review]
In the HTML lab I added, I used the format `lab-recipe-page` for the block dashed name and filenames - it doesn't sound as good, but it keeps all the labs next to each other in the file tree - the URL will be `lab-recipe-page` though, so there's some trade-offs. The display name is still set in the intro.json file, so we can show whatever we want.

I think we should be consistent, so we should either change this to use that format, or I can change the other one. What do you think @zairahira?

File: client/i18n/locales/english/intro.json [hunk 1]
```diff
@@ -1699,6 +1699,12 @@
       "front-end-development-certification-exam": {
         "title": "Front End Development Certification Exam",
         "intro": ["", ""]
+      },
+      "lab-business-card": {
+        "title": "Design a Business Card",
+        "intro": [
+          "In this lab, you'll create a business card and style it using CSS"
```
Comment (Dario-DC) [inline client/i18n/locales/english/intro.json:1706]:
```suggestion
          "In this lab, you'll create a business card and style it using CSS."
```

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 2]
```diff
@@ -0,0 +1,468 @@
+---
+id: 6690e10ebe2181212abc9652
+title: Design a Business Card
+challengeType: 14
+dashedName: lab-business-card
+---
+
+# --description--
+
+**Objective:** Design a Business card that looks similar to <a href="https://codepen.io/rancid_beet/full/ExBaJdw" target="_blank" rel="noopener noreferrer nofollow">https://business-card.freecodecamp.rocks</a>. Fulfill the user stories below and get all the tests to pass to complete the lab.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:10]:
```suggestion
**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
```
To keep the same format in all labs.

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+assert.strictEqual(new __helpers.CSSHelp(document).getStyle('body').getPropVal('font-family'), 'Arial, sans-serif');
+```
+
+Your `p` elements should have a top and bottom margin of `5px` and a left and right margin of `0px`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('p').getPropVal('margin'), '5px 0px');
+```
+
+The links of the page should have no underline by default
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:49]:
```suggestion
The links of the page should have no underline by default.
```

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+assert.exists(document.querySelector('img.profile-image'));
+```
+
+The image source should be set to a valid image.
+
+```js
+assert.match(document.querySelector('.profile-image').getAttribute('src'),/.+/);
+```
+
+The `alt` attribute of the image should be set to a meaningful description.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:123]: It always pass.

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+assert.exists(document.querySelector('div.business-card p.designation'));
+```
+
+After the `designation` `p` element, there should be a `p` element with the class `company`.
+
+```js
+assert.exists(document.querySelector('div.business-card p.company'));
+```
+
+The first `p` element with the class `full-name` should contain your name.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:166]: It always pass.

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+assert.equal(new __helpers.CSSHelp(document).getStyle('.full-name').getPropVal('font-size'), '24px');
+```
+
+The font color of the first `p` element with the class `full-name` should be `darkslategray`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.full-name').getPropVal('color'), 'darkslategray');
+```
+
+The second `p` element with the class `designation` should contain your designation.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:184]: It always pass

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 7]
```diff
@@ -0,0 +1,468 @@
...
+assert.equal(new __helpers.CSSHelp(document).getStyle('.designation').getPropVal('font-size'), '18px');
+```
+
+The font color of the second `p` element with the class `designation` should be `gray`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.designation').getPropVal('color'), 'gray');
+```
+
+The third `p` element with the class `company` should contain your company name.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:202]: It always pass

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 8]
```diff
@@ -0,0 +1,468 @@
...
+assert.equal(new __helpers.CSSHelp(document).getStyle('hr').getPropVal('margin'), '20px 0px');
+```
+
+The `hr` elements should have a top and bottom margin of `20px` and a left and right margin of `0px`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('hr').getPropVal('margin'), '20px 0px');
+```
+
+After the first `hr` element, there should be a `p` element with an email address as your text.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:244]: It always pass

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 9]
```diff
@@ -0,0 +1,468 @@
...
+assert.match(document.querySelectorAll(".business-card p")[4].textContent,/.+/);
+```
+
+After the phone number `p` element, there should be an `a` element with the text `Portfolio` 
+
+```js
+assert.equal(document.querySelector('.business-card a').textContent, 'Portfolio');
+```
+
+Your portfolio `a` element should have a valid `href`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:262]: It always pass

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 10]
```diff
@@ -0,0 +1,468 @@
...
+assert.equal(document.querySelector('.social-media h2').textContent, 'Connect with me');
+```
+
+Inside the `.social-media` class, there should be an `a` element with the text `Twitter` 
+
+```js
+assert.equal(document.querySelectorAll(".social-media a")[0].textContent, 'Twitter');
+```
+
+The Twitter `href` should point to a valid link.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:316]: It always pass

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 11]
```diff
@@ -0,0 +1,468 @@
...
+```
+
+Inside the `.social-media` class, there should be an `a` element with the text `LinkedIn`.
+
+```js
+
+assert.equal(document.querySelectorAll(".social-media a")[1].textContent, 'LinkedIn');
+```
+
+The LinkedIn `href` should point to a valid link.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:331]: It always pass

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 12]
```diff
@@ -0,0 +1,468 @@
...
+assert.match(linkedInA?.href, /.+/);
+```
+
+Inside the `.social-media` class, there should be an `a` element with the text `GitHub`
+
+```js
+assert.equal(document.querySelectorAll(".social-media a")[2].textContent, 'GitHub');
+```
+
+The GitHub `href` should point to a valid link.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:345]: It always pass

**Review by Dario-DC — COMMENTED** [review]
I just had a brief look. There are several tests that pass without adding any code.

File: curriculum/challenges/_meta/lab-business-card/meta.json [hunk 13]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "Design a Business Card",
+  "isUpcomingChange": true,
+  "usesMultifileEditor": true,
+  "dashedName": "lab-business-card",
+  "order": 2,
```
Comment (moT01) [inline curriculum/challenges/_meta/lab-business-card/meta.json:6]: The issue about the order should be fixed, so you should be able to change this to whatever.

Comment (moT01) [reply curriculum/challenges/_meta/lab-business-card/meta.json:6]: Also, I updated the spreadsheet to fill in the block order - it looks like this one is 38

**Review by moT01 — CHANGES_REQUESTED** [review]
I was able to follow the instructions and pass all the tests, so I think I would be fine with this 👍 I do have a number of suggestions. ..

Since this is the first CSS project, we could maybe add a user story to "link the styles.css file". I don't know if it's needed, but I couldn't figure out why my styles weren't being applied for a bit. Maybe users will be used to it at this point.

I personally think the user stories and tests could be trimmed down quite a bit, and allow for more flexibility. For instance, many of the elements ask for a specific background color - similarly with font-size and other properties. This makes it so many user projects will end up looking very similar - I would prefer to have some more flexibility. Maybe just one or two where you give something specific and then allow anything in other areas - or not even include a user story if it's been covered. For instance, you could say "The body should have a background color of rosybrown." and then maybe "other thing should have a background color set to something other than the default." Or with the font-size, just have them set a font-size for one or two things, and leave out the others - and they can set it to anything if they want to or just leave it. The goal of implementing font-size would be accomplished.

I don't think that stuff needs to be added here, I'm just explaining how I think it could be better - kind of just a preference.

All that said, I did have fun doing it and it covers what it needs to 👍

**Comment by zairahira** [comment]
Thank you @moT01 for a detailed review. 

I have made the following changes:

- Added a story and tests for linking style sheet to HTML.
- Trimmed the user stories and adjusted the tests accordingly.
- Fixed the failing test for Firefox.
- Improved vague instructions/ tests.

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 14]
```diff
@@ -0,0 +1,418 @@
+---
+id: 6690e10ebe2181212abc9652
+title: Design a Business Card
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:3]:
```suggestion
title: Build a Business Card
```

File: curriculum/challenges/_meta/lab-business-card/meta.json [hunk 15]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "Build a Business Card",
+  "isUpcomingChange": true,
+  "usesMultifileEditor": true,
+  "dashedName": "lab-business-card",
+  "order": 38,
+  "superBlock": "front-end-development",
+  "challengeOrder": [{ "id": "6690e10ebe2181212abc9652", "title": "Design a Business Card" }],
```
Comment (Dario-DC) [inline curriculum/challenges/_meta/lab-business-card/meta.json:8]:
```suggestion
  "challengeOrder": [{ "id": "6690e10ebe2181212abc9652", "title": "Build a Business Card" }],
```

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 16]
```diff
@@ -0,0 +1,418 @@
...
+assert.equal(new __helpers.CSSHelp(document).getStyle('.business-card').getPropVal('margin-left'), 'auto');
+```
+
+Inside the `.business-card` class, there should be an `img` element with the class `profile-image`.
+
+```js
+assert.exists(document.querySelector('img.profile-image'));
+```
+
+The image source should be set to a valid image.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:151]: Once `img.profile-image`  is created this is always satisfied, even if you don't set the `src`.
I would check the length of the attribute instead.

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 17]
```diff
@@ -0,0 +1,418 @@
...
+assert.match(document.querySelector('.profile-image').getAttribute('src'),/.+/);
+```
+
+The `alt` attribute of the image should be set to a meaningful description.
+
+```js
+assert.isAbove($('img')[0].alt.length, 0);
+```
+
+Inside the `.business-card` class, after the `img` element, there should be a `p` element with the class `full-name`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:163]:
```suggestion
Inside the `.business-card` element, after the `img` element, there should be a `p` element with the class `full-name`.
```

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 18]
```diff
@@ -0,0 +1,418 @@
...
+assert.exists(document.querySelector('a + hr'));
+```
+
+After the second `hr` element, there should be a `div` element with the class `social-media`.
+
+```js
+assert.exists(document.querySelector('div.business-card div.social-media'));
+```
+
+Inside the `.social-media` class, there should be an `h2` element with the text `Connect with me`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:247]:
```suggestion
Inside the `.social-media` element, there should be an `h2` element with the text `Connect with me`.
```

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 19]
```diff
@@ -0,0 +1,418 @@
...
+
+```js
+assert.equal(document.querySelector('.social-media h2').textContent, 'Connect with me');
+```
+
+Inside the `.social-media` class, there should be an `a` element with the text `Twitter`
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:253]:
```suggestion
Inside the `.social-media` element, there should be an `a` element with the text `Twitter`.
```

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 20]
```diff
@@ -0,0 +1,418 @@
...
+assert.equal(document.querySelectorAll(".social-media a")[0].textContent, 'Twitter');
+```
+
+The Twitter `href` should point to a valid link.
+
+```js
+assert.isAbove($('.social-media a')[0].getAttribute('href').length, 0);
+```
+
+Inside the `.social-media` class, there should be an `a` element with the text `LinkedIn`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:265]:
```suggestion
Inside the `.social-media` element, there should be an `a` element with the text `LinkedIn`.
```

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 21]
```diff
@@ -0,0 +1,418 @@
...
+assert.equal(document.querySelectorAll(".social-media a")[1].textContent, 'LinkedIn');
+```
+
+The LinkedIn `href` should point to a valid link.
+
+```js
+assert.isAbove($('.social-media a')[1].getAttribute('href').length, 0);
+```
+
+Inside the `.social-media` class, there should be an `a` element with the text `GitHub`
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:278]:
```suggestion
Inside the `.social-media` element, there should be an `a` element with the text `GitHub`.
```

**Review by Dario-DC — COMMENTED** [review]
The project is looking good :+1:  I left a few comments.

Since this is about styling, I would also check that the image size is appropriate. Set its width should be enough.

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md (hunk 14 above)
Comment (moT01) [reply curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:3]: Do we want to strictly use "Build a (project)"? I kinda liked "Design" here, but I'm maybe we want to just stick with "Build" for consistency.

Comment (jdwilkin4) [reply curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:3]: I was thinking about this and here are my two cents

For all HTML, JS, React /Typescript, and Backend labs it should use the Build naming convention.
For select CSS labs, where appropriate, it is fine if it is Design X naming convention

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 22]
```diff
@@ -0,0 +1,418 @@
...
+
+**User Stories:**
+
+1. You should link the style sheet `styles.css` using the `link` tag in the `head` section of the HTML file.
+1. Your page background should be `rosybrown`. The overall font should be `Arial`, with a fallback of `sans-serif`.   
+1. All of the `p` elements should have a top and bottom margin of `5px`.
+1. All links on the page should have a default of no underline. When you hover over the links, there should be an underline.
+1. You should have a `div` with the class `business-card`. 
+1. The `business-card` class should be `300px` wide and it should have any background color of your choice. There should be a padding of `20px` all around and a top margin of `100px`. The text should be center aligned and the font size should be `16px`. The left and right margins should be set to `auto`.
+1. Inside the `buisness-card` div, there should be an `img` element with the class `profile-image`. You can set the image source to `https://cdn.freecodecamp.org/curriculum/labs/flower.jpg` if you like. There should be an `alt` with a meaningful description.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:20]:
```suggestion
1. Inside the `business-card` div, there should be an `img` element with the class `profile-image`. You can set the image source to `https://cdn.freecodecamp.org/curriculum/labs/flower.jpg` if you like. There should be an `alt` with a meaningful description.
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md (hunk 16 above)
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:152]: It should work now.

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 23]
```diff
@@ -0,0 +1,419 @@
...
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should link the style sheet `styles.css` using the `link` tag in the `head` section of the HTML file.
+1. Your page background should be `rosybrown`. The overall font should be `Arial`, with a fallback of `sans-serif`.   
+1. All of the `p` elements should have a top and bottom margin of `5px`.
+1. All links on the page should have a default of no underline. When you hover over the links, there should be an underline.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:18]:
```suggestion
1. By default, all links on the page should not be underlined. When you hover over the links, they should be underlined.
```
The first sentence implied (to me) that "no underline" was a property that the links could have.

**Review by ojeytonwilliams — APPROVED** [review]
Hey @zairahira, since I'll be away for a bit and don't want to be in the way, I've left an approval.

I did notice one minor thing, though, but it's not blocking, imo.

**Review by jdwilkin4 — APPROVED** [review]
LGTM 🚀

File: curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md [hunk 24]
```diff
@@ -0,0 +1,419 @@
...
+id: 6690e10ebe2181212abc9652
+title: Design a Business Card
+challengeType: 14
+dashedName: design-a-business-card
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. **Do not copy this demo project**
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:11]:
```suggestion
**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
```
As I understand we are not going to add that in the description of each project.

Comment (moT01) [reply curriculum/challenges/english/25-front-end-development/lab-business-card/6690e10ebe2181212abc9652.md:11]: I'm going to commit this so we can get this merged. I don't think any of the other projects have it. We can go through and make sure they are consistent later as well.

**Review by moT01 — APPROVED** [review]

**Pull request merged**